tools/underwater-gps.py

```diff
@@ -184,79 +276,86 @@ def processLocatorPosition(response, *args, **kwargs):
     result['vdop'] = 1.0
     result['satellites_visible'] = 10
     result['ignore_flags'] = 8 | 16 | 32
-    result = json.dumps(result);
-    print('sending      ', result)
-
-    sockit.sendto(result, ('0.0.0.0', 25100))
-
-def notifyPutResponse(response, *args, **kwargs):
-    print('PUT response:', response.text)
-
-update_period = 0.25
-last_master_update = 0
-last_locator_update = 0
-s = grequests.Session()
-# Thank you https://stackoverflow.com/questions/16015749/in-what-way-is-grequests-asynchronous
-while True:
-    if time.time() > last_locator_update + update_period:
-        last_locator_update = time.time()
-        url = gpsUrl + "/api/v1/position/global"
-        print('requesting data from', url)
-        request = grequests.get(url, session=s, hooks={'response': processLocatorPosition})
-        grequests.send(request)
-
-    if time.time() > last_master_update + update_period:
-        last_master_update = time.time()
-        url = gpsUrl + "/api/v1/position/master"
-        print('requesting data from', url)
-        request = grequests.get(url, session=s, hooks={'response': processMasterPosition})
-        grequests.send(request)
-
-    try:
-        datagram = sockit.recvfrom(4096)
-        recv_payload = json.loads(datagram[0])
-        if 'depth' not in recv_payload or 'temp' not in recv_payload:
-            print('not valid payload:')
-            print(json.dumps(recv_payload, indent=4, sort_keys=True))
+    result = json.dumps(result)
+    socket_mavproxy.sendto(result, ('0.0.0.0', 25100))
+
+
+# Socket to send GPS data to mavproxy
+socket_mavproxy = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+socket_mavproxy.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+socket_mavproxy.setblocking(0)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Driver for the Water Linked Underwater GPS system.")
+    parser.add_argument('--ip', action="store", type=str, default="demo.waterlinked.com", help="remote ip to query on.")
+    parser.add_argument('--port', action="store", type=str, default="80", help="remote port to query on.")
+    args = parser.parse_args()
+
+    # Use UDP port 14401 to send NMEA data to QGC for topside location
+    qgcNmeaSocket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
```

Comment:
there is variables that are snake and others that are camel, please be consistent